### PR TITLE
refine TiFlash doc to add executor pushdown white list

### PR DESCRIPTION
**Pull request by windtalker**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

refine TiFlash doc to add executor pushdown white list
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by ilovesoup — APPROVED** [review]
LGTM

File: tiflash/use-tiflash.md [hunk 1]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
-目前下推连接 (`Join`) 的功能需要通过以下会话变量开启（暂不支持 `Full Outer Join`）：
-
-```
-set @@session.tidb_opt_broadcast_join=1
-```
-
-目前 TiFlash 支持了有限的常用表达式下推，支持下推的表达式包括：
+TiFlash 支持部分算子的下推，具体支持的算子包括：
+
+* TableScan：该算子从表中读取数据
```
Comment (zanmato1984) [inline tiflash/use-tiflash.md:222]: 是不是还有sort和apply算子不能下推？

File: tiflash/use-tiflash.md [hunk 2]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+    * 仅支持不带 group by 列的 StreamAgg
+* TopN：该算子对数据求 top n 运算
+* Limit：该算子对数据进行 limit 运算
+* Project：该算子对数据进行投影运算
+* HashJoin：该算子基于 [Hash Join](/explain-joins.md/#Hash Join) 算法对数据进行连接运算
+    * 只有在 [MPP 模式](#使用 MPP 模式)下才能下推
+    * 必须带有等值的 join 条件
+    * `Full Outer Join` 不支持下推
+
+大部分算子中都包含有表达式计算，当且仅当一个算子所包含的所有表达式均支持下推给 TiFlash 时，该算子才有可能下推给 TiFlash。目前 TiFlash 支持下推的表达式包括：
```
Comment (zanmato1984) [inline tiflash/use-tiflash.md:235]: 可以再提一下这种情况：下层不能下推的算子挡在中间导致上层可以下推的算子也无法下推。

File: tiflash/use-tiflash.md [hunk 3]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
-TiFlash 支持谓词、聚合下推计算以及表连接，下推的计算可以帮助 TiDB 进行分布式加速。暂不支持的计算类型是 `Full Outer Join` 和 `DISTINCT COUNT`，会在后续版本逐步优化。
-
-目前下推连接 (`Join`) 的功能需要通过以下会话变量开启（暂不支持 `Full Outer Join`）：
-
-```
-set @@session.tidb_opt_broadcast_join=1
-```
-
-目前 TiFlash 支持了有限的常用表达式下推，支持下推的表达式包括：
+TiFlash 支持部分算子的下推，具体支持的算子包括：
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:220]:
```suggestion
TiFlash 支持部分算子的下推，支持的算子如下：
```

File: tiflash/use-tiflash.md [hunk 4]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+
+* TableScan：该算子从表中读取数据
+* Selection：该算子对数据进行过滤
+* HashAgg：该算子基于 [Hash Aggregation](/explain-aggregation.md/#Hash Aggregation) 算法对数据进行聚合运算
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:224]:
```suggestion
* HashAgg：该算子基于 [Hash Aggregation](/explain-aggregation.md#hash-aggregation) 算法对数据进行聚合运算
```

File: tiflash/use-tiflash.md [hunk 5]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+* StreamAgg：该算子基于 [Stream Aggregation](/explain-aggregation.md/#Stream Aggregation) 算法对数据进行聚合运算
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:225]:
```suggestion
* StreamAgg：该算子基于 [Stream Aggregation](/explain-aggregation.md#stream-aggregation) 算法对数据进行聚合运算。StreamAgg 仅支持不带 `GROUP BY` 条件的列。
```

File: tiflash/use-tiflash.md (hunk 1 above)
Comment (windtalker) [reply tiflash/use-tiflash.md:222]: 是的，不过这里说的是支持的算子，不支持的算子没列，而且不支持的除了sort和apply，还有别的其他算子

Comment (zanmato1984) [reply tiflash/use-tiflash.md:222]: 哦哦，抱歉我搞错了。。。

File: tiflash/use-tiflash.md (hunk 2 above)
Comment (windtalker) [reply tiflash/use-tiflash.md]: 已更新

File: tiflash/use-tiflash.md [hunk 6]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+    * 仅支持不带 group by 列的 StreamAgg
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:226]:
```suggestion
```

File: tiflash/use-tiflash.md [hunk 7]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+* TopN：该算子对数据求 top n 运算
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:227]:
```suggestion
* TopN：该算子对数据求 TopN 运算
```

File: tiflash/use-tiflash.md [hunk 8]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+* Limit：该算子对数据进行 limit 运算
+* Project：该算子对数据进行投影运算
+* HashJoin：该算子基于 [Hash Join](/explain-joins.md/#Hash Join) 算法对数据进行连接运算
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:230]:
```suggestion
* HashJoin：该算子基于 [Hash Join](/explain-joins.md#hash-join) 算法对数据进行连接运算，但有以下使用条件：
```

File: tiflash/use-tiflash.md [hunk 9]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+    * 只有在 [MPP 模式](#使用 MPP 模式)下才能下推
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:231]:
```suggestion
    * 只有在 [MPP 模式](#使用 MPP 模式)下才能被下推
```

File: tiflash/use-tiflash.md [hunk 10]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+    * 必须带有等值的 join 条件
+    * `Full Outer Join` 不支持下推
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:233]:
```suggestion
    * 不支持下推 `Full Outer Join`
```

File: tiflash/use-tiflash.md [hunk 11]
```diff
@@ -217,15 +217,22 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+
+在 TiDB 中算子之间会组织成树的形式，一个算子能下推给 TiFlash 的前提条件是该算子的所有孩子算子都能下推给 TiFlash，而且因为大部分算子中都包含有表达式计算，当且仅当一个算子所包含的所有表达式均支持下推给 TiFlash 时，该算子才有可能下推给 TiFlash。目前 TiFlash 支持下推的表达式包括：
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:235]:
```suggestion
在 TiDB 中，算子之间会呈现树型组织结构。一个算子能下推到 TiFlash 的前提条件，是该算子的所有子算子都能下推到 TiFlash。因为大部分算子都包含有表达式计算，当且仅当一个算子所包含的所有表达式均支持下推到 TiFlash 时，该算子才有可能下推给 TiFlash。目前 TiFlash 支持下推的表达式包括：
```

File: tiflash/use-tiflash.md [hunk 12]
```diff
@@ -234,12 +241,9 @@ round without fraction, cast(int as decimal), date_add(datetime, int), date_add(
 
 其中，`cast` 和 `date_add` 的下推默认不开启，若需要手动开启，请参考[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)
 
-目前 TiFlash 不支持下推的情况包括：
-
-- 所有包含 Time 类型的表达式均不能下推
-- 在聚合函数或者 WHERE 条件中包含了不在上述列表中的表达式，聚合或者相关的谓词过滤均不能下推
+另外目前 TiFlash 对于所有包含 Time/Bit/Set/Enum/Geometry 类型的表达式均不能下推
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:244]:
```suggestion
另外，所有包含 Time/Bit/Set/Enum/Geometry 类型的表达式均不能下推到 TiFlash：
```

File: tiflash/use-tiflash.md (hunk 11 above)
Comment (qiancai) [inline tiflash/use-tiflash.md:235]:
```suggestion
在 TiDB 中，算子之间会组织成树的形式，一个算子能下推给 TiFlash 的前提条件是该算子的所有子算子都能下推给 TiFlash，而且因为大部分算子中都包含有表达式计算，当且仅当一个算子所包含的所有表达式均支持下推给 TiFlash 时，该算子才有可能下推给 TiFlash。目前 TiFlash 支持下推的表达式包括：
```

**Review by zanmato1984 — APPROVED** [review]
LGTM

File: tiflash/use-tiflash.md (hunk 6 above)
Comment (windtalker) [reply tiflash/use-tiflash.md:226]: 这个为什么要删掉？这应该是个限制条件

File: tiflash/use-tiflash.md [hunk 13]
```diff
@@ -234,12 +240,9 @@ round without fraction, cast(int as decimal), date_add(datetime, int), date_add(
 
 其中，`cast` 和 `date_add` 的下推默认不开启，若需要手动开启，请参考[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)
 
-目前 TiFlash 不支持下推的情况包括：
-
-- 所有包含 Time 类型的表达式均不能下推
-- 在聚合函数或者 WHERE 条件中包含了不在上述列表中的表达式，聚合或者相关的谓词过滤均不能下推
+另外，所有包含 Time/Bit/Set/Enum/Geometry 类型的表达式均不能下推到 TiFlash：
```
Comment (qiancai) [inline tiflash/use-tiflash.md:243]:
```suggestion
另外，所有包含 Time/Bit/Set/Enum/Geometry 类型的表达式均不能下推到 TiFlash。
```

File: tiflash/use-tiflash.md [hunk 14]
```diff
@@ -234,12 +240,9 @@ round without fraction, cast(int as decimal), date_add(datetime, int), date_add(
 
 其中，`cast` 和 `date_add` 的下推默认不开启，若需要手动开启，请参考[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)
```
Comment (qiancai) [inline tiflash/use-tiflash.md:241]:
```suggestion
其中，`cast` 和 `date_add` 的下推默认不开启，若需要手动开启，请参考[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)。
```

File: tiflash/use-tiflash.md [hunk 15]
```diff
@@ -234,12 +240,9 @@ round without fraction, cast(int as decimal), date_add(datetime, int), date_add(
...
 
-目前 TiFlash 不支持下推的情况包括：
-
-- 所有包含 Time 类型的表达式均不能下推
-- 在聚合函数或者 WHERE 条件中包含了不在上述列表中的表达式，聚合或者相关的谓词过滤均不能下推
+另外，所有包含 Time/Bit/Set/Enum/Geometry 类型的表达式均不能下推到 TiFlash：
 
-如查询遇到不支持的下推计算，则需要依赖 TiDB 完成剩余计算，可能会很大程度影响 TiFlash 加速效果。对于暂不支持的表达式，将会在后续陆续加入支持，也可以联系官方沟通。
+如查询遇到不支持的下推计算，则需要依赖 TiDB 完成剩余计算，可能会很大程度影响 TiFlash 加速效果。对于暂不支持的算子/表达式，将会在后续陆续加入支持，也可以联系官方沟通。
```
Comment (qiancai) [inline tiflash/use-tiflash.md:245]:
```suggestion
如查询遇到不支持的下推计算，则需要依赖 TiDB 完成剩余计算，可能会很大程度影响 TiFlash 加速效果。对于暂不支持的算子和表达式，将会在后续版本中陆续支持，也可以联系官方沟通。
```

**Comment by TomShawn** [comment]
Follow-up: resolve qiancai's comments.

**Comment by TomShawn** [comment]
label requires-followup

**Pull request merged**